python/setup.py

```diff
@@ -38,7 +38,7 @@ def scan_for_package_data(
     "jsonschema>=4.21.1,<5",
     "sentry-sdk>=2.0.0",
     "pysher==1.0.8",
-    "pydantic>=2.6.4,<2.10",
+    "pydantic>=2.6.4",
```

Comment:
Removing the upper bound constraint for pydantic could potentially lead to compatibility issues with future versions. Consider:
1. Adding a comment in setup.py explaining why we're comfortable removing the upper bound
2. Document any testing done with newer pydantic versions
3. Add a note in the changelog about this change